.github/workflows/rust.yml

```diff
@@ -1,37 +1,32 @@
-name: Rust
+name: Test Rust package
 
 on:
   # always trigger on PR
   push:
+    branches:
+      - main
   pull_request:
   # manual trigger
   # https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow
   workflow_dispatch:
 
+env:
+  # Disable full debug symbol generation to speed up CI build and keep memory down
+  # "1" means line tables only, which is useful for panic tracebacks.
+  RUSTFLAGS: "-C debuginfo=1"
+
 jobs:
   # Check crate compiles
   linux-build-lib:
     name: cargo check
     runs-on: ubuntu-latest
-    container:
-      image: amd64/rust
```

Comment:
Shouldn't need to use this container as stable Rust is already included in the GHA runners